### ISSUE TITLE
feat: Training log doesn't show environment ini vars

### ISSUE THREAD

**Issue by sjseth** [sjseth/AI-Case-Sorter-Py#100]
### Before you submit

- [x] This is a feature request, not a support or usage question
- [x] I have searched existing issues and no duplicate exists

### Problem description

Previous versions show information about whether we are using GPU, CPU and if CUDA was initialized. This is important information for someone who happens to be waiting for a long training to be able to articulate. 

### Desired solution

When starting a training, this information should be output in the status window to the top of the training log to be consistent with previous versions and the windows version. The training log should also be logged to the logs folder and accessible in the "Get Support" section or at least some easy way for the user to capture and send the log in discord

---------

[SETUP] Found existing PyTorch setup (gpu_modern mode).
CUDA: True

[INFO] Device: CUDA (RTX OPTIMIZED)
[INFO] Detected GPU: NVIDIA GeForce RTX 5060 Ti 16GB

[INFO] Configuration:
       Model: convnext_small
       Batch Size: 8
       Initial LR: 0.0002
       Weight Decay: 0.0001
       Dropout: 0.0
       Validation Split: 0.2
       Full Dataset Training: True
       Image Size: 480x480
       Focal Loss: False
       Stochastic Depth: torchvision default
       SWA: False
       Target Epochs: 12

[INFO] Loaded 42 images. Classes: 13
[INFO] --trainall specified -> using full dataset for training
[INFO] Stochastic depth prob: torchvision default
[INFO] channels_last memory format: enabled (device capability sm_120)
[INFO] Using CrossEntropyLoss with label smoothing

[Train] 1/12: 100%|#######################################| 6/6 [00:24<00:00,  4.01s/it, acc=0.1667]
[INFO] Epoch 1 Train Loss: 2.5273, Train Acc: 0.1667, LR: 0.000197
